Source/Core/DolphinQt/MenuBar.cpp

```diff
@@ -506,7 +506,12 @@ void MenuBar::AddViewMenu()
   AddShowRegionsMenu(view_menu);
 
   view_menu->addSeparator();
-  view_menu->addAction(tr("Purge Game List Cache"), this, &MenuBar::PurgeGameListCache);
+  QAction* const purge_action =
+      view_menu->addAction(tr("Purge Game List Cache"), this, &MenuBar::PurgeGameListCache);
+  connect(&Settings::Instance(), &Settings::GameListRefreshRequested, purge_action,
+          [purge_action] { purge_action->setEnabled(false); });
+  connect(&Settings::Instance(), &Settings::GameListRefreshCompleted, purge_action,
+          [purge_action] { purge_action->setEnabled(true); });
```

Comment:
The "Purge" action isn't disabled on launch. Clicking it does nothing initially until the list completely loads then it purges of course.
You'll need to cause "GameListRefreshCompleted" to be emitted on launch though. I stopped looking for the ideal means for this to happen. You're probably more familiar with this code.

Reply:
Bonus points for never disabling "Purge" and just making it cancel and restart the current operation.

Reply:
Let's see if it possible to abort, dispose the pending events in the working thread, and then restart, so there is no need to enable/disable the button/action.

Reply:
Unfortunately, it was not straight forward to implement your second proposal.

For now, I addressed the problem you initially mentioned; the button/action are now disabled on launch until the GameTracker is started.

While I think it should be possible to make the button/action more responsive, it involves a bit more fiddling than I first thought.

It is trivial to flush the WorkerQueueThread object before enqueuing the purge/refresh event, but there are other IO operations that can be noticeably blocking. I artificially added some delay to some of these operations to verify that flushing the event queue is not enough; things that are currently being processed also need to be halted (when possible). This means we still need to disable/enable the button/action, or else the user might feel the lack of responsiveness (but hopefully the button/action should be re-enabled much more promptly).

I guess I will prepare a new PR _soon_.